### PR TITLE
Add create and create_or_throw factories for Compression_Algorithm and Decompression_Algorithm.

### PR DESCRIPTION
**Pull request by lambdafu**
I noticed that make_compressor was working differently from the create/create_or_throw factories I already knew for hash functions. Assuming that create/create_or_throw is the newer interface, I added them for compression and decompression (I guess adding an alias CompressionFunction = Compression_Algorithm, and DecompressionFunction = Decompression_Algorithm is also interesting, but I didn't do that).

If this doesn't go in the right direction, please disregard it. :)

### CODE REVIEW

File: src/lib/compression/compression.h [hunk 1]
```diff
@@ -19,6 +19,22 @@ namespace Botan {
 class BOTAN_PUBLIC_API(2,0) Compression_Algorithm
    {
    public:
+      /**
+      * Create an instance based on a name, or return null if the
+      * algo/provider combination cannot be found. If provider is
+      * empty then best available is chosen.
```
Comment (randombit) [inline src/lib/compression/compression.h]: No provider argument here so this statement is bogus.

File: src/lib/compression/compression.h [hunk 2]
```diff
@@ -65,6 +81,22 @@ class BOTAN_PUBLIC_API(2,0) Compression_Algorithm
 class BOTAN_PUBLIC_API(2,0) Decompression_Algorithm
    {
    public:
+      /**
+      * Create an instance based on a name, or return null if the
+      * algo/provider combination cannot be found. If provider is
+      * empty then best available is chosen.
```
Comment (randombit) [inline src/lib/compression/compression.h]: Same

**Comment by randombit** [comment]
Thanks! Minor nit on the Doxygen comments (references non-existent provider argument) other than that LGTM

**Comment by lambdafu** [comment]
Sorry that I missed those.  I remember removing two other ones :cloud:

**Pull request merged**